### PR TITLE
add advancedsetting for ssl ca trust bundle fixes #19006

### LINKED ISSUE #19006: Can not set my own cacert.pem on android without root
### Describe the bug
Here is a clear and concise description of what the problem is:

On an Android system system there is no option to use your own cacert.pem file as described in https://kodi.wiki/view/SSL_certificates
There is actually no way to set the environment SSL_CERT_FILE on your phone for example.

The problem is also described on the forums several times: 
https://forum.kodi.tv/showthread.php?tid=345878
https://forum.kodi.tv/showthread.php?tid=350721
https://forum.kodi.tv/showthread.php?tid=345894

It seams that SSL_CERT_FILE gets set to special://xbmc/system/certs/cacert.pem which from the log get's set to /data/user/0/org.xbmc.kodi/cache/apk/assets which is not writable without root access

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

If I place a file in kodi home system/certs/cacert.pem I would expect kodi to use that file instead of the build in one. 
That way I can upload my cacert.pem file and place it in /storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/system/certs/cacert.pem and solve the problem on any system

## Actual Behavior

Placing certs/cacert.pem file in the kodi's home system directory has no effect

### To Reproduce

1. 
2.
3.

### Debuglog

The debuglog can be found here:
https://pastebin.com/LHmhRDnN

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [x] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version:

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by wsnipex**
## Description

adds an advanded setting to override the SSL CA trust bundle

usage:
```
  <network>
    <catrustfile>special://masterprofile/cacerts.pem</catrustfile>
  </network>
```

## Motivation and Context

On certain platforms, e.g. android, it's not currently possible to use a custom CA trust bundle.

## How Has This Been Tested?

Only on linux

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/settings/AdvancedSettings.h [hunk 1]
```diff
@@ -288,6 +288,8 @@ class CAdvancedSettings : public ISettingCallback, public ISettingsHandler
     bool m_curlDisableIPV6;
     bool m_curlDisableHTTP2;
 
+    std::string m_catrustfile;
```
Comment (garbear) [inline xbmc/settings/AdvancedSettings.h:291]: Can you use camelCase? My cat rust file won't stop chasing corrosive data.

Comment (wsnipex) [reply xbmc/settings/AdvancedSettings.h]: done

Comment (garbear) [reply xbmc/settings/AdvancedSettings.h]: I would also change `cacert`. Um... my cace is realtime?

**Review by arnova — APPROVED** [review]

**Comment by Tolriq** [comment]
Beware that in a not so far future Kodi won't be able to read files from arbitrary path and will require either SAF or files to be inside specific Kodi directories.
Maybe better long term solution would be to have a fallback read from a writable user directory inside Kodi like special://home or special://userdata (To be checked where it points on Android)

**Review by garbear — APPROVED** [review]

**Comment by wsnipex** [comment]
updated to support special:// paths as well, but I don't want to restrict it to a specific path.

**Comment by arnova** [comment]
lgtm

**Pull request merged**